bin/reth/src/node/mod.rs

```diff
@@ -229,9 +229,25 @@ impl Command {
         }
 
         info!(target: "reth::cli", "Connecting to P2P network");
+        let secret_key_path = self.p2p_secret_key.clone().unwrap_or_default();
         let default_secret_key_path = data_dir.p2p_secret_path();
+
+        let mut secret_key_path = secret_key_path.as_path();
+        if secret_key_path.to_str() == Some("") {
+            secret_key_path = default_secret_key_path.as_path();
+        }
```

Comment:
what's the reasoning for this here?

Reply:
The current code always returns the default secret_key_path and does NOT return the --p2p-secret-key parameter if set.
My changes use "p2p-secret-key" if specified, else, use default value.

Reply:
sorry, perhaps I misunderstood 

so the actual bug here is that the user input is ignored?

> silently reverts to default location if file is not found

Reply:
how exactly does your change fix

> Setup Reth so that the default location for the p2p secret key is not accessible (e.g. systemd ProtectHome=True).
> Set --p2p-secret-key to a non existent file.
> This results in the following very confusing failure:

Reply:
Problems Address here:
1) User input is ignored.
This is a new problem introduced the the last couple days that I ran across and tried to fix here.
Blame Commit:
feat: use chain-specific data dirs (#2495)
https://github.com/paradigmxyz/reth/commit/ecc7ae9bb111d07506aa56655bac9d6c8136d1d9

2) Original problem
The --p2p-path-key set by user.
If the file does not exist, then the default_path is silently used/created without erroring.

This PR Addresses 2 cases:
1) If p2p-secret-key IS set, try to load it, if it doesn't exist, create it.
2) If p2p-secret-key is NOT set, use default path, if it doesn't exist create it.